wurst/objects/abilities/beastMaster/SummonBeast.wurst

```diff
@@ -0,0 +1,90 @@
+package SummonBeast
+
+// Standard Library Imports:
+import AbilityObjEditing
+import ObjectIds
+import Assets
+
+// Local Imports:
+import LocalObjectIDs
+import ElkDefinition
+import JungleWolfDefinition
+import JungleBearDefinition
+
+let ICON = Icons.bTNStag
+let TOOLTIP = "Summon a beast to fight along side you"
+let TOOLTIP_EXT = "Each level will increase the ferocity of the animal fighting along your side"
+
+let DAMAGE_BASE = 7
+let DAMAGE_RANGE = 3
+let DAMAGE_PER_LEVEL = 3
+let COOLDOWN = 60.
+let DURATION = 30.
+
+class ElkLocust extends AdultElkAttackerDefinition
```

Comment:
I don't see the point of using a class type per ability definition and have been trying to move away from that, like how @Jaccouille did in #668

Reply:
We realized it wasn't good design as it lacked flexibility and might end up with "big" constructor prototype which were hard to read, [this](https://github.com/island-troll-tribes/island-troll-tribes/blob/cc1bc2981f5f7c11c3bdf91e9c73fee74b62f247/wurst/objects/units/TrollBuildings.wurst) is an example of what @jonathanmclaus came with, you end up with more line but it's easier to update if you need to alter an attribute for a specific object definition.

Reply:
I don't mind switching it, seems like a reasonable conclusion. But before spending any more time on this idea we should determine which course of action we want to take on this rework.

Reply:
I think it works better than the current system, so still worth merging it even if it's temporary.

Reply:
If you want to talk about course of action, you might want to describe the context in the PR, because I don't understand :
1. What is the current issue
2. How does this solve the issue or what are the expected result, iirc it's an ability who spawns a locust unit for X seconds, which attack nearby enemies, I could label it as "uncontrolled damage source" and I kind of fail to see where the fun part in playing this.

Reply:
@Jaccouille , I think the core problems are (in my opinion):

1. Bear is a big power spike, as it is stronger and not really balanced.
2. Any pet makes killing hostiles trivial, as they have free healing and revive.
3. You can have the pet kill items on the enemy island with minimal risk. Even if the enemy finds it, they have to waste their time killing it and again, this costs the BM nothing.

I do think that is not as fun, but it solves all of these immediately and still is relevant to the class theme. I'd like to address these better in the long run though.

Reply:
Why not nerf bear stats and have pets give XP on death?

Reply:
Like Jon said, right now it's not balanced, and generally I think it would be difficult to balance Shapeshifter having a pet with how powerful the beast forms should be. Thus, in order to have ability continuity from the base class to the subclass, I would like to switch to the locust abilities. Yes, this might make BM a little more right-clicky, but items give you so much interactivity that I don't think this is a problem. I plan on keeping pets in their current form on Druid, and eventually I'll work on improving the pet system for Druid/Rendo.

Reply:
Still believe this is less fun/interactive than #569 but you're free to test w/e you want as I trust you have bigger plan.

Reply:
@Jaccouille your PR can still be merged into the plan for Druid, wouldn't you agree?

Reply:
I am not against this suggestion and would gladly spent some time to adjust/think more about it, however I am on a tight schedule at this moment.